packages/starlight/integrations/virtual-user-config.ts

```diff
@@ -10,15 +10,15 @@ function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
 /** Vite plugin that exposes Starlight user config and project context via virtual modules. */
 export function vitePluginStarlightUserConfig(
 	opts: StarlightConfig,
-	{ root, srcDir }: Pick<AstroConfig, 'root' | 'srcDir'>
+	{ root, srcDir, build }: Pick<AstroConfig, 'root' | 'srcDir' | 'build' >
 ): NonNullable<ViteUserConfig['plugins']>[number] {
 	const resolveId = (id: string) =>
 		JSON.stringify(id.startsWith('.') ? resolve(fileURLToPath(root), id) : id);
 
 	/** Map of virtual module names to their code contents as strings. */
 	const modules = {
 		'virtual:starlight/user-config': `export default ${JSON.stringify(opts)}`,
-		'virtual:starlight/project-context': `export default ${JSON.stringify({ root, srcDir })}`,
+		'virtual:starlight/project-context': `export default ${JSON.stringify({ root, srcDir, buildFormat: build.format })}`,
```

Comment:
I’d be tempted to keep this mirroring Astro’s config shape 1:1:

```suggestion
		'virtual:starlight/project-context': `export default ${JSON.stringify({ root, srcDir, build: { format: build.format } })}`,
```